src/currencyConversion/CurrencyConversionAPI.ts

```
@@ -34,7 +34,7 @@ export default class CurrencyConversionAPI<TContext = any> extends DataSource {
     currencyCode,
     timestamp,
     impliedExchangeRates,
-  }: CurrencyConversionArgs): Promise<BigNumber> {
+  }: CurrencyConversionArgs): Promise<BigNumber | null> {
```

I think it's better to throw an exception instead of returning `null` or `undefined`.
So when using this service we don't have to consider both cases (exception or `null` value) as a not normal execution. When probably we would handle both cases as the same.

---

The same applies for other services.